src/zulip.rs

```diff
@@ -92,11 +93,18 @@ struct Response {
 /// Top-level handler for Zulip webhooks.
 ///
 /// Returns a JSON response or a 400 with an error message.
-// TODO: log JsonRejection
 pub async fn webhook(
     State(ctx): State<Arc<Context>>,
-    Json(req): Json<Request>,
+    req: Result<Json<Request>, JsonRejection>,
 ) -> axum::response::Response {
+    let Json(req) = match req {
```

Comment:
Does this actually happen in practice, that Zulip sends us invalid JSON?

Reply:
No, but invalid incorporate de-serialization failures, which is possible since we have our own structs.

And as we saw with https://rust-lang.zulipchat.com/#narrow/channel/224082-triagebot/topic/Team.20API.20not.20accessible.20in.20GHA.20log.20viewer, it's better to have details.